pom.xml

```diff
@@ -86,12 +87,13 @@
     <hbase.suffix>hbase-${hbase.profile}</hbase.suffix>
 
     <!-- This is used by the release script only -->
-    <hbase.profile.list>2.1 2.2 2.3 2.4</hbase.profile.list>
+    <hbase.profile.list>2.1 2.2 2.3 2.4 2.4.1</hbase.profile.list>
```

Comment:
> In this case, we should rename the profile names, as 2.4 working with 2.4.0 instead of the latest version is counter-intuitive.

Agree, sounds good.

> The other soultion, that I originally had in mind was to map the profiles to the latest supported patch level, and only build official binaries for those.

Hmm, although this is doable but for latest releases, it might not be in our best interest to make users stick to a particular patch release? 
On the other hand, we could make decisions on the basis of the what urgency does HBase patch release bring? For instance, 2.4.1 has some critical authorization fixes, which made 2.4.1 to come out way ahead of the schedule. For such urgent releases, it might be good to support them sooner, but for non-critical patch release that cause compatibility issues in Phoenix, we might want to wait little longer.

Anyways, if you are fine with supporting first approach (new profile for 2.4.1), I will also raise another Jira to remove HBase 2.1 support in Phoenix 5.2. One less profile for 5.2 for now. WDYT?

Moreover, I understand the pain with ever increasing profiles. As far as running tests are concerned in multibranch, perhaps, we could limit it to latest 4 profiles?

Reply:
Our response time to any new incompatible HBase release is not really relevant to this issue. I think that doing a new Phoenix (patch) release ASAP after such a release would help us be more relevant in the community.

Thinking a bit more about this, I propose the following:

- We add new profiles for each compatibility module, as you did, they are cheap, and reduce the possibility of errors
- The patclevel-less profiles always point to the latest supported HBase compatibility module and HBase patch release (i.e. profile 2.4 uses HBase 2.4.1, and profile 2.4.0 uses HBase 2.4.0) (though the compat-module version and the HBase versions are usually different, as the compati-modules version is the first supported, but we build with the éast supported HBase release)
- Profiles for older patchlevels are named after the compat-module release
- We update BULDING.md to reflect the above
- We run jenkins tests only for latest profiles (the patchlevel-less ones)

We discuss whether we want to build release artifacts for the artifacts on the list. 
(If you are fine with the above, you could start the discussion)

Reply:
As for removing HBase 2.1 support, I'd try to stick to the unofficial 1 year+1 release after EOL suggested by Xinyi, which I believe has not elapsed yet.

Reply:
> The patclevel-less profiles always point to the latest supported HBase compatibility module and HBase patch release (i.e. profile 2.4 uses HBase 2.4.1, and profile 2.4.0 uses HBase 2.4.0)

I think this sounds better. And this is needed only for incompatible patch release, otherwise for compatible patch release we would just bump up `hbase.version` on pom.xml directly.

Reply:
We call it **hbase-<profile>.runtime.version** now, but yes.